Messenger/FBMApplication.m

```diff
@@ -4,12 +4,31 @@
 @implementation FBMApplication
 
 - (void)sendEvent:(NSEvent *)event {
-  // Cmd-1 .. Cmd-9
-  if (event.type == NSKeyDown && (event.modifierFlags & NSCommandKeyMask) && event.keyCode >= 18 && event.keyCode <= 28) {
-    [((AppDelegate*)self.delegate) setActiveConversationAtIndex:event.characters];
-    return;
+
+  if (event.type == NSKeyDown) {
+    if (event.modifierFlags & NSCommandKeyMask) {
+
+      // Cmd-1 .. Cmd-9
+      if (event.keyCode >= 18 && event.keyCode <= 28) {
+        [((AppDelegate*)self.delegate) setActiveConversationAtIndex:event.characters];
+        return;
+      }
+
+      // Cmd-N to begin a new message
+      else if (event.keyCode == 0x2D) {
+        [((AppDelegate*)self.delegate) beginNewMessage];
+        return;
+      }
+    }
+
+    // If Tab is pressed from the new conversation "To:" box, we should focus on the textbox, just like in OSX Messages
+    else if (event.keyCode == 0x30 &&
```

Comment:
Note: keyCode is hardware-dependent. You should inspect the event's `charactersIgnoringModifiers` property instead. If you pull and merge with upstream master, this code has been refactored to do just that.